### PR TITLE
fixing flaky testInnerJoin() test

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/CDAP-6391
Build: http://builds.cask.co/browse/CDAP-DUT4343-6

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java [hunk 1]
```diff
@@ -85,7 +87,9 @@
   static final Type SINK_OUTPUTS_TYPE = new TypeToken<Map<String, SinkOutput>>() { }.getType();
   private static final Logger LOG = LoggerFactory.getLogger(ETLMapReduce.class);
   private static final Gson GSON = new GsonBuilder()
-    .registerTypeAdapter(SetMultimap.class, new SetMultimapCodec<>()).create();
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .registerTypeAdapter(SetMultimap.class, new SetMultimapCodec<>())
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java:91]: Isn't it necessary to enable complex map type serialization, via the GsonBuilder?

Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java:91]: I think its only needed when we have map key as complex object. In our case, we have map value as complex object (Schema).

Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java:91]: oh yeah; that's true.
I guess the value is automatically handles complex values.

Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java:91]: Thats right.

**Comment by anwar6953** [comment]
lgtm

**Comment by CuriousVini** [comment]
Ran Bamboo build 3 times on this branch. The build is green all 3 times. So merging.

**Pull request merged**